front/src/Phaser/Entity/Character.ts

```diff
@@ -159,6 +159,19 @@ export abstract class Character extends Container implements OutlineableInterfac
         return { x: this.x, y: this.y };
     }
 
+    /**
+     * Returns position based on where player is currently facing
+     * @param shift How far from player should the point of interest be.
+     */
+    public getDirectionalActivationPosition(shift: number): { x: number, y: number } {
+        switch (this.lastDirection) {
+            case PlayerAnimationDirections.Down: { return { x: this.x, y: this.y + shift }; }
+            case PlayerAnimationDirections.Left: { return { x: this.x - shift, y: this.y }; }
+            case PlayerAnimationDirections.Right: { return { x: this.x + shift, y: this.y }; }
+            case PlayerAnimationDirections.Up: { return { x: this.x, y: this.y - shift }; }
+        }
```

Comment:
When it comes to enums, I usually like to end my switch/case statements with a default statement that cannot compile (because I'm casting to the "never" type that cannot be attributed).

```suggestion
        switch (this.lastDirection) {
            case PlayerAnimationDirections.Down: { return { x: this.x, y: this.y + shift }; }
            case PlayerAnimationDirections.Left: { return { x: this.x - shift, y: this.y }; }
            case PlayerAnimationDirections.Right: { return { x: this.x + shift, y: this.y }; }
            case PlayerAnimationDirections.Up: { return { x: this.x, y: this.y - shift }; }
            default: {
                    // Security check: if we forget a "case", the line below will catch the error at compile-time.
                    const tmp: never = this.lastDirection;
                }
        }
```

Typescript will happily compile this, but if one day, we decide to add other directions (for instance if we add isometric tiles), the switch statement won't compile anymore, with a nice error message.

Reply:
I think it is not really needed as TypeScript will throw you an error if not all of enum's cases are covered:
`Function lacks ending return statement and return type does not include 'undefined'.`